crates/subspace-farmer/src/utils/parity_db_store.rs

```diff
@@ -1,21 +1,28 @@
 use parity_db::{ColumnOptions, Db, Options};
 use std::error::Error;
 use std::fmt;
+use std::fmt::Debug;
 use std::marker::PhantomData;
 use std::path::Path;
 use std::sync::Arc;
-use subspace_core_primitives::Piece;
-use subspace_networking::libp2p::kad::record::Key;
 use tracing::{debug, trace, warn};
 
 /// Generic key value store with ParityDB backend and iteration support
 #[derive(Clone)]
-pub struct ParityDbStore {
+pub struct ParityDbStore<StoreKey, StoreValue> {
     // Parity DB instance
     db: Arc<Db>,
+
+    // Type marker
+    marker: PhantomData<(StoreKey, StoreValue)>,
 }
 
-impl ParityDbStore {
+impl<StoreKey, StoreValue> ParityDbStore<StoreKey, StoreValue>
+where
+    StoreKey: AsRef<[u8]> + From<Vec<u8>> + Debug,
```

Comment:
I'm not so sure about `From<Vec<u8>>`, it feels like we should do scale encoding here instead, though it will be backwards incompatible change. Might be fine as is for now, but I can easily imagine the type for which canonical `From<Vec<u8>>` doesn't make much sense, yet we might want to use it as a key.

BTW, do we need `From<Vec<u8>>` at all? Looks like we already have `TryFrom<Vec<u8>>` below and it should only be constrained to `iter()` method, not the whole thing.